### PR TITLE
feat: Import keys PoC & new server req/res types

### PR DESCRIPTION
**Pull request by Danziger**
- [x] Update API request format and use B64 serialization.
- [x] Import keys PoC code (unused/disabled). 
- [x] Improve legacy buffer types/interfaces.
- [x] Update API response format.
- [x] Update CI to stop publishing to NPM automatically and instead just run tests and the build. 
- [x] Update `sign()` and `signDataItem()` to propagate all params from the `Transaction` passed in as param.
- [x] Close https://github.com/Othent/KeyManagementService/issues/39.
- [x] Add new `__overridePublicKey()` and `__getServerInfo()` functions for development/debugging.

### CODE REVIEW

**Review by 7i7o — APPROVED** [review]
tACK 👍

File: src/lib/auth/auth0.ts [hunk 1]
```diff
@@ -36,10 +30,15 @@ import {
...
 import { getAnsProfile } from "../utils/ans/ans.utils";
 import { PROVIDER_LABELS } from "./auth0.constants";
+import { B64UrlString, uint8ArrayTob64Url } from "../utils/arweaveUtils";
+import { transactionInputReplacer } from "./auth0.utils";
+import { pemToUint8Array } from "../othent-kms-client/operations/importKey";
 
 export class OthentAuth0Client {
   private debug = false;
 
+  private overriddenPublicKey: B64UrlString | null = null;
```
Comment (Danziger) [inline src/lib/auth/auth0.ts:40]: This can be useful for development/debugging.

File: src/lib/othent-kms-client/operations/server-info.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  const encodedData = options?.includeToken ? await auth0.encodeToken() : null;
+
+  let serverInfo: ServerInfo | null = null;
+
+  try {
+    const serverInfoResponse = encodedData
+      ? await api.post<ServerInfo>(Route.HOME, {
+          encodedData,
+        } satisfies CommonEncodedRequestData)
+      : await api.get<ServerInfo>(Route.HOME);
```
Comment (Danziger) [inline src/lib/othent-kms-client/operations/server-info.ts:34]: Note the current server only supports the `GET` option here (the handler can take care of both `GET` and `POST` but I've just noticed I forgot to add the route). In any case, as this is an "internal" functionality for development, we can release the SDK change already and update the server later.

File: src/lib/othent/othent.ts [hunk 3]
```diff
@@ -949,17 +965,29 @@ export class Othent implements Omit<ArConnect, "connect"> {
 
     // This function returns a new signed transaction. It doesn't mutate/sign the original one:
     const transactionToSign = await arweave.createTransaction({
```
Comment (Danziger) [inline src/lib/othent/othent.ts:967]: I added all the missing `Transaction` properties that we were not using before in here.

File: src/lib/othent/othent.ts [hunk 4]
```diff
@@ -1314,4 +1374,14 @@ export class Othent implements Omit<ArConnect, "connect"> {
 
     return Promise.resolve(this.tokens.has(id));
   }
+
+  // DEVELOPMENT:
```
Comment (Danziger) [inline src/lib/othent/othent.ts:1378]: Added these 2 for development.

**Review by 7i7o — APPROVED** [review]
ACK 👍

**Pull request merged**